File: Rapid React 0000/src/main/java/frc/robot/commands/OpIntake.java
package frc.robot.commands;

//Motor Dependancies
import com.ctre.phoenix.motorcontrol.ControlMode;

//Import WPI
import edu.wpi.first.wpilibj2.command.CommandBase;

//Import Robot Files
import frc.robot.subsystems.IntakeSubsystem;
import frc.robot.Robot;


public class OpIntake extends CommandBase {
    private IntakeSubsystem intake;

    public OpIntake(final IntakeSubsystem intakeSubSys){
        intake = intakeSubSys;
        addRequirements(intake);
    }

    @Override
    public void initialize(){
        intake.startCompressor();
        intake.raise(true, true);
    }

    @Override
    public void execute(){
        
        if(Robot.io.master.getLeftBumperPressed()){
            intake.raise(true, true);
        }
        else if(Robot.io.master.getRightBumperPressed()){
            intake.lower(true, true);
        };

        if(Robot.io.master.getBButton()){
            intake.set(ControlMode.PercentOutput, -1);
        }
        else if(intake.getFrontState() == true){
            intake.setFront(ControlMode.PercentOutput, 1);
        }
        else{
            intake.setFront(ControlMode.PercentOutput, 0);
        };

        if(Robot.io.master.getXButton()){
            intake.set(ControlMode.PercentOutput, -1);
        }
        else if(intake.getRearState() == true){
            intake.set(ControlMode.PercentOutput, 1);
        }
        else{
            intake.setRear(ControlMode.PercentOutput, 0);
        }

    }

    @Override
    public void end(boolean interrupted){
        intake.stopCompressor();
        intake.stop(false);
    }

    @Override
    public boolean isFinished(){
        return false;
    }
}

File: 6364 Worlds/src/main/cpp/commands/IntakeCommands/OpIntake.cpp
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

#include "commands/IntakeCommands/OpIntake.hpp"
#include "RobotContainer.h"
#include <frc/DriverStation.h>


IntakeCommands::OpIntake::OpIntake(IntakeSubsystem* SubSystem_intake) : intake{SubSystem_intake} {
    AddRequirements(intake);
}


void IntakeCommands::OpIntake::Initialize(){
        if(compressor.Enabled() == false){
        printf("Intake Error: Compressor Not Enabled\n");
            if(compressor.GetPressureSwitchValue() == false){
                printf("Tank Pressure Nominal");
            }
        }
        else if(compressor.GetPressureSwitchValue()){
            printf("WARNING: Tank Pressure Low\n");
        }
}

void IntakeCommands::OpIntake::Execute(){

    if(master.GetPOV(0) == 270 || partner.GetPOV(0) == 270){
        intake->raise(false, true);
    }
    else if(master.GetPOV(0) == 90 || partner.GetPOV(0) == 90){
        intake->raise(true, false);
    }

    if(master.GetRightTriggerAxis() > 0.1 || partner.GetRightTriggerAxis() > 0.1){
        intake->setFront(ControlMode::PercentOutput, 1);
        intake->lower(true, false);
    }
    else if(master.GetXButton()){
        intake->setFront(ControlMode::PercentOutput, -1);
        intake->lower(true, false);
    }
    else{
        intake->setFront(ControlMode::PercentOutput, 0);
    }

    if(master.GetLeftTriggerAxis() > 0.1 || partner.GetLeftTriggerAxis() > 0.1){
        intake->setRear(ControlMode::PercentOutput, 1);
        intake->lower(false, true);  
    }
    else if(master.GetYButton()){
        intake->setRear(ControlMode::PercentOutput, -1);
        intake->lower(false, true);
    }
    else{
        intake->setRear(ControlMode::PercentOutput, 0);
    }
}

void IntakeCommands::OpIntake::End(bool interrupted){
    intake->setFront(ControlMode::PercentOutput, 0);
    intake->setRear(ControlMode::PercentOutput, 0);
    intake->raise();
}

bool IntakeCommands::OpIntake::IsFinished(){
    return false;
}